rust/src/transformations/make_stable_expr/expr_sum/mod.rs

```diff
@@ -0,0 +1,308 @@
+where
+    MI: 'static + UnboundedMetric,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+    Expr: StableExpr<PartitionDistance<MI>, PartitionDistance<MI>>,
+{
+    let Expr::Agg(AggExpr::Sum(prior_expr)) = expr else {
+        return fallible!(MakeTransformation, "expected function expression");
+    };
+
+    let t_prior = prior_expr.make_stable(input_domain, input_metric)?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let dtype = &middle_domain.active_series()?.field.dtype;
+
+    // check that we are in a context where it is ok to break row-alignment
+    middle_domain.context.break_alignment()?;
+
+    // build output domain
+    let mut output_domain = middle_domain.clone();
+
+    // summation invalidates bounds on the active column
+    output_domain.active_series_mut()?.drop_bounds()?;
+
+    // we only care about the margin that matches the current grouping columns
+    let margin_id = middle_domain.context.grouping_columns()?;
+    let input_margin = (output_domain.frame_domain.margins.remove(&margin_id))
+        .ok_or_else(|| err!(MakeTransformation, "failed to find margin {:?}", margin_id))?;
+
+    // Set the margins on the output domain to consist of only one margin: 1 row per group, with at most 1 record in each group.
+    let output_margin = input_margin.clone().with_max_partition_length(1);
+    output_domain.frame_domain.margins = HashMap::from_iter([(margin_id, output_margin)]);
+
+    fn stability_map<MI, const P: usize, TI>(
+        series_domain: SeriesDomain,
+        margin: Margin,
+    ) -> Fallible<StabilityMap<PartitionDistance<MI>, LpDistance<P, f64>>>
+    where
+        MI: UnboundedMetric,
+        TI: Summand,
+        f64: InfCast<TI::Sum> + InfCast<u32>,
+    {
+        let (l, u) = series_domain.atom_domain::<TI>()?.get_closed_bounds()?;
+        let (l, u) = (TI::Sum::neg_inf_cast(l)?, TI::Sum::inf_cast(u)?);
+
+        let public_info = margin
+            .public_info
+            .ok_or_else(|| err!(MakeTransformation, "keys must be public information"))?;
+
+        let max_size = usize::exact_int_cast(margin.max_partition_length.ok_or_else(|| {
+            err!(
+                MakeTransformation,
+                "must specify max_partition_length in margin"
+            )
+        })?)?;
+
+        let pp_relaxation = f64::inf_cast(TI::Sum::relaxation(max_size, l, u)?)?;
+
+        let norm_map = move |d_in: f64| match P {
+            1 => Ok(d_in),
+            2 => d_in.inf_sqrt(),
+            _ => return fallible!(MakeTransformation, "unsupported Lp norm"),
+        };
+
+        let pp_map = move |d_in: &IntDistance| match public_info {
+            MarginPub::Keys => TI::Sum::inf_cast(*d_in)?.inf_mul(&l.alerting_abs()?.total_max(u)?),
+            MarginPub::Lengths => TI::Sum::inf_cast(*d_in / 2)?.inf_mul(&u.inf_sub(&l)?),
+        };
+
+        Ok(StabilityMap::new_fallible(
+            move |(l0, l1, li): &(IntDistance, IntDistance, IntDistance)| {
```

Comment:
perhaps
```suggestion
            move |(l0, l1, l_inf): &(IntDistance, IntDistance, IntDistance)| {
```

Worry about the possibility of misreading `1` and `i`